cmd/starbridge/integrations/my_ethereum_contract.go

```diff
@@ -4,9 +4,17 @@ import (
 	"math/big"
 	"strings"
 
+	supportlog "github.com/stellar/go/support/log"
 	"github.com/stellar/starbridge/cmd/starbridge/model"
 )
 
+// global for now
+var logger *supportlog.Entry
+
+func SetLogger(l *supportlog.Entry) {
+	logger = l
+}
```

Comment:
Instead of creating global/package level state, could you create a type that is instantiated with its dependencies, where one of those dependencies is the logger?

Opinion pieces on this:
- https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html
- https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables
- https://leighmcculloch.com/posts/tool-go-check-no-globals-no-inits/

Example of how some of the other packages avoid package state:

https://github.com/stellar/starbridge/blob/cd4215d914fb90d287779c96ce7370a5fd8660e9/cmd/starbridge/sigsharestellar/sigsharestellar.go#L12-L42

Reply:
yes, that's how I generally like to do things too :)
Just kept this format temporarily for execution speed. I will be doing a larger reorg of the code after our demo tomorrow to make these improvements among a few others